balancer/randomsubsetting/randomsubsetting.go

```diff
@@ -0,0 +1,201 @@
+	Name = "random_subsetting"
+)
+
+var (
+	logger = grpclog.Component(Name)
+)
+
+func prefixLogger(p *subsettingBalancer) *internalgrpclog.PrefixLogger {
+	return internalgrpclog.NewPrefixLogger(logger, fmt.Sprintf("[random-subsetting-lb %p] ", p))
+}
+
+func init() {
+	balancer.Register(bb{})
+}
+
+type bb struct{}
+
+func (bb) Build(cc balancer.ClientConn, bOpts balancer.BuildOptions) balancer.Balancer {
+	b := &subsettingBalancer{
+		cc:    cc,
+		hashf: xxhash.NewWithSeed(uint64(time.Now().UnixNano())),
+	}
+	// Create a logger with a prefix specific to this balancer instance.
+	b.logger = prefixLogger(b)
+
+	b.logger.Infof("Created")
+	b.child = gracefulswitch.NewBalancer(cc, bOpts)
+	return b
+}
+
+// LBConfig is the config for the outlier detection balancer.
+type LBConfig struct {
+	serviceconfig.LoadBalancingConfig `json:"-"`
+
+	SubsetSize uint64 `json:"subset_size,omitempty"`
+
+	ChildPolicy *iserviceconfig.BalancerConfig `json:"child_policy,omitempty"`
+}
+
+func (bb) ParseConfig(s json.RawMessage) (serviceconfig.LoadBalancingConfig, error) {
+	lbCfg := &LBConfig{
+		// Default top layer values.
+		SubsetSize: 10,
+	}
+
+	if err := json.Unmarshal(s, lbCfg); err != nil { // Validates child config if present as well.
+		return nil, fmt.Errorf("subsetting: unable to unmarshal LBconfig: %s, error: %v", string(s), err)
+	}
+
+	// if someonw needs subsetSize == 1, he should use pick_first instead
+	if lbCfg.SubsetSize < 2 {
+		return nil, errors.New("subsetting: subsetSize must be >= 2")
+	}
+
+	return lbCfg, nil
+}
+
+func (bb) Name() string {
+	return Name
+}
+
+type subsettingBalancer struct {
+	cc     balancer.ClientConn
+	logger *internalgrpclog.PrefixLogger
+	cfg    *LBConfig
+	hashf  *xxhash.Digest
+	child  *gracefulswitch.Balancer
+}
+
+func (b *subsettingBalancer) UpdateClientConnState(s balancer.ClientConnState) error {
+	lbCfg, ok := s.BalancerConfig.(*LBConfig)
+	if !ok {
+		b.logger.Errorf("received config with unexpected type %T: %v", s.BalancerConfig, s.BalancerConfig)
+		return balancer.ErrBadResolverState
+	}
+
+	// Reject whole config if child policy doesn't exist, don't persist it for
+	// later.
+	bb := balancer.Get(lbCfg.ChildPolicy.Name)
+	if bb == nil {
+		return fmt.Errorf("subsetting: child balancer %q not registered", lbCfg.ChildPolicy.Name)
+	}
+
+	if b.cfg == nil || b.cfg.ChildPolicy.Name != lbCfg.ChildPolicy.Name {
+		err := b.child.SwitchTo(bb)
+		if err != nil {
+			return fmt.Errorf("subsetting: error switching to child of type %q: %v", lbCfg.ChildPolicy.Name, err)
+		}
+	}
+	b.cfg = lbCfg
+
+	err := b.child.UpdateClientConnState(balancer.ClientConnState{
+		ResolverState:  b.prepareChildResolverState(s.ResolverState),
+		BalancerConfig: b.cfg.ChildPolicy.Config,
+	})
+
+	return err
+}
+
+type AddressWithHash struct {
+	hash uint64
+	addr resolver.Address
+}
+
+// implements the subsetting algorithm, as described in A68: https://github.com/grpc/proposal/pull/423
```

Comment:
Nit: A68 is now merged. So, you can actually link to the doc in the repo instead of linking to the PR. Also, you could do that at the top-level package docstring instead.